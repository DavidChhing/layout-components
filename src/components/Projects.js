import Component from './Component';
import styles from './Projects.css';


export default class Projects extends Component {
  render() {

    const projectContent = this.props.projects.map(project => {
      return (
        `<h3>${project.name}</h3>
        <p> ${project.details} </p>`
      )
    });

    return `<section class="${styles.projectSection}">
    <h3>Projects</h3>
    <div id="projects">
    ${projectContent }
    </div>
    `;
  };
}
