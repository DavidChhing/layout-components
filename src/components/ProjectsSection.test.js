import Projects from "./Projects";

describe("Projects component", () => {
  it("should be a project section", () => {
    const projects = new Projects({
      projects: [],
    });
    expect(projects.render()).toMatchSnapshot();
  });
});
