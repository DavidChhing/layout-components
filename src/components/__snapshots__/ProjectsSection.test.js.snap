// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects component should be a project section 1`] = `
"<section class=\\"projectSection\\">
    <h3>Projects</h3>
    <div id=\\"projects\\">
    
    </div>
    "
`;
